Chap_API_Server.tex

```diff
@@ -2411,18 +2411,19 @@ \subsection{\code{pmix_server_dmodex_req_fn_t}}
 %%%%
 \descr
 
-Used by the \ac{PMIx} server to request its local host contact the \ac{PMIx} server on the remote node that hosts the specified proc to obtain and return any information that process posted via calls to \refapi{PMIx_Put} and \refapi{PMIx_Commit}.
+Used by the \ac{PMIx} server to request its local host contact the \ac{PMIx} server on the remote node that hosts the specified proc to obtain (using \refapi{PMIx_server_dmodex_request}) and return any information that the process, \refarg{proc}, posted via calls to \refapi{PMIx_Put} and \refapi{PMIx_Commit}.
 
 The array of \refarg{info} structs is used to pass user-requested options to the server.
 This can include a timeout to preclude an indefinite wait for data that may never become available.
 The directives are optional unless the \emph{mandatory} flag has been set - in such cases, the host \ac{RM} is required to return an error if the directive cannot be met.
 
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \subsubsection{Dmodex attributes}
 
 %
 \declareAttribute{PMIX_REQUIRED_KEY}{"pmix.req.key"}{char*}{
-Identifies a key that must be included in the requested information. If the specified key is not already available, then the \ac{PMIx} servers are required to delay response to the dmodex request until either the key becomes available or the request times out.
+Identifies a key that must be included in the requested information. If the specified key is not already available, then the \ac{PMIx} servers are required to delay response to the dmodex request until either the key becomes available or the request times out.  The server can test the availability of a required key by calling \refapi{PMIx_Get} on the key.  Since there is no mechanism to retract a key-value that has been made available by a client through PMIx_Put, a successful \refapi{PMIx_Get} call on the key by the server is sufficient to ensure that a subsquent \refapi{PMIx_server_dmodex_request} call targeting the process will include the key.
```

Comment:
```suggestion
Identifies a key that must be included in the requested information. If the specified key is not already available, then the \ac{PMIx} servers are required to delay response to the dmodex request until either the key becomes available or the request times out.  The server can test the availability of a required key by calling \refapi{PMIx_Get} on the key.  Since there is no mechanism to retract a key-value that has been made available by a client through PMIx_Put, a successful \refapi{PMIx_Get} call on the key by the server is sufficient to ensure that a subsequent \refapi{PMIx_server_dmodex_request} call targeting the process will include the key.
```

I think I'd like this to be an advise, given that it hints at a possible implementation without necessarily mandating it.